src/Distribution/Server/Features/BuildReports/BuildReport.hs

```diff
@@ -57,6 +59,10 @@ import Distribution.CabalSpecVersion
          ( CabalSpecVersion(CabalSpecV2_4) )
 import Distribution.Pretty
          ( Pretty(..), pretty, prettyShow )
+#if MIN_VERSION_Cabal(3,7,0)
+import Distribution.Fields.Pretty
+         ( pattern NoComment )
+#endif
```

Comment:
Alternatively, instead of `#if`, we could burn the bridges and drop Cabal-3.6 entirely.